cmd/swarm/swarm-snapshot/create.go

```diff
@@ -0,0 +1,181 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/simulations"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/network/simulation"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+const testMinProxBinSize = 2
+const NoConnectionTimeout = 1
+
+func create(ctx *cli.Context) error {
+	log.PrintOrigins(true)
+	log.Root().SetHandler(log.LvlFilterHandler(log.Lvl(verbosity), log.StreamHandler(os.Stdout, log.TerminalFormat(true))))
+
+	if len(ctx.Args()) < 1 {
+		return errors.New("argument should be the filename to verify or write-to")
+	}
+	filename, err := touchPath(ctx.Args()[0])
+	if err != nil {
+		return err
+	}
+	err = discoverySnapshot(filename, 10)
+	if err != nil {
+		utils.Fatalf("Simulation failed: %s", err)
+	}
+
+	return err
+}
+
+func discoverySnapshot(filename string, nodes int) error {
+	log.Debug("discoverySnapshot", "filename", filename, "nodes", nodes)
+	//disable discovery if topology is specified
+	discovery = topology == ""
+	ids := make([]enode.ID, 0)
+	sim := simulation.New(map[string]simulation.ServiceFunc{
+		"bzz": func(ctx *adapters.ServiceContext, b *sync.Map) (node.Service, func(), error) {
+			addr := network.NewAddr(ctx.Config.Node())
+
+			kp := network.NewKadParams()
+			kp.MinProxBinSize = testMinProxBinSize
+
+			kad := network.NewKademlia(addr.Over(), kp)
+			hp := network.NewHiveParams()
+			hp.KeepAliveInterval = time.Duration(200) * time.Millisecond
+			hp.Discovery = discovery
+
+			log.Info(fmt.Sprintf("discovery for nodeID %s is %t", ctx.Config.ID.String(), hp.Discovery))
+
+			config := &network.BzzConfig{
+				OverlayAddr:  addr.Over(),
+				UnderlayAddr: addr.Under(),
+				HiveParams:   hp,
+			}
+			ids = append(ids, ctx.Config.ID)
+			return network.NewBzz(config, kad, nil, nil, nil), nil, nil
+
+		},
+	})
+	defer sim.Close()
+
+	_, err := sim.AddNodes(10)
+	if err != nil {
+		utils.Fatalf("%v", err)
+	}
+
+	events := make(chan *simulations.Event)
+	sub := sim.Net.Events().Subscribe(events)
+	select {
+	case ev := <-events:
+		//only catch node up events
+		if ev.Type == simulations.EventTypeConn {
+			utils.Fatalf("this shouldn't happen as connections weren't initiated yet")
+		}
+	case <-time.After(NoConnectionTimeout * time.Second):
+	}
+
+	sub.Unsubscribe()
+
+	if len(sim.Net.Conns) > 0 {
+		utils.Fatalf("no connections should exist after just adding nodes")
+	}
+
+	switch topology {
```

Comment:
i totally do not see a need fpr this at all,

Reply:
i'm not sure i understand

Reply:
we just dont need this. please remove

Reply:
just to confirm, this infers that discovery should always be _on_ for creating a snapshot. correct?